src/commandargs.cpp

```diff
@@ -5,16 +5,27 @@
 
 static ProgramConfiguration* Config = nullptr;
 static Setting* CurrentState = nullptr;
+String programName;
 
 bool MatchesFlag(String token)
 {
+	if(token.length() < 1)
+	{
+		return false;
+	}
+
     for(Setting& s: *Config)
     {
         if(token == s.Flag)
         {
             CurrentState = &s;
             return true;
         }
+        else if(token.at(0) == '-')
+        {
+        	// If an unknown which resembles a flag is passed, inform the user
+        	Usage(std::vector<SettingOption>());
```

Comment:
I like this though.

Reply:
A possible improvement to this is to specify the unknown flag, but there's not a lot of flags here, so that's kind of a moot point